docker/Dockerfile

```diff
@@ -25,12 +25,14 @@ RUN apk add --no-cache --virtual .build-dependencies \
 COPY ./pyproject.toml ./poetry.lock ./
 
 ARG dev
+ARG simulate_removal
 RUN apk add --no-cache --virtual .build-dependencies \
     gcc \
     git \
+    linux-headers \
     musl-dev \
     && \
     poetry install ${dev} && \
-    apk del .build-dependencies
+    apk del ${simulate_removal} .build-dependencies
```

Comment:
Long story short: if the `dev` image removes the packages in `.build-dependencies`, `poetry install` may break as part of `make add-dependency` as it doesn't find `git` or can't compile other packages.